src/SWP/Bundle/ContentBundle/Factory/ArticleFactory.php

```diff
@@ -74,8 +76,13 @@ public function create()
      */
     public function createFromPackage(PackageInterface $package)
     {
+        /** @var ArticleInterface $article */
         $article = $this->create();
-        $article->setParentDocument($this->articleProvider->getParent($this->contentRelativePath));
+
+        if ($article instanceof HierarchyInterface) {
```

Comment:
Are we going to try to support both orm and phpcr? I think we should decide one way or another. Personally, I am against it, but if we are going to do it we should probably do it properly, with tests everywhere, etc. Otherwise any code to support this will be redundant, as no one will use it.